apps/hash/README.md

```diff
@@ -378,6 +386,68 @@ New packages which are to be built as JavaScript, whether as an app or dependenc
 
+### Authoring Patches
+
+Patches to JavaScript packages are managed by Yarn, using the [`yarn patch`](https://yarnpkg.com/cli/patch) command.
+
+#### Creating a new patch
+
+```sh
+yarn patch <package>
+# ➤ YN0000: Package <package>@npm:<version> got extracted with success!
+# ➤ YN0000: You can now edit the following folder: /private/var/folders/lk/j93xz9pd7nqgd5_2wlyxmbh00000gp/T/xfs-df787c87/user
+# ➤ YN0000: Once you are done run yarn patch-commit -s /private/var/folders/lk/j93xz9pd7nqgd5_2wlyxmbh00000gp/T/xfs-df787c87/user and Yarn will store a patchfile based on your changes.
+# ➤ YN0000: Done in 0s 702ms
+```
+
+Once you have completed your changes, run the command that was output to commit the patch:
+
+```sh
+yarn patch-commit -s /private/var/folders/lk/j93xz9pd7nqgd5_2wlyxmbh00000gp/T/xfs-df787c87/user
+```
+
+This will automatically create a patch file and put it into the `.yarn/patches` directory. If you're modifying a direct dependency in any workspace it will replace the `package.json` entry with a `patch:` reference to the patch file. In case you're patching an indirect dependency a new resolutions entry will be added to the root workspace `package.json`.
+
+You will need to run `yarn install` for the patch to be installed and applied to the lockfile.
+
+#### Modifying an existing patch
+
+The procedure to modify an existing patch is very similar, but instead of running `yarn patch <package>` you will need to run `yarn patch -u <package>`. This will apply existing patches and then extract the package for you to modify.
+
+```sh
+yarn patch -u <package>
+# ➤ YN0000: Package <package>@npm:<version> got extracted with success along with its current modifications!
+# ➤ YN0000: You can now edit the following folder: /private/var/folders/lk/j93xz9pd7nqgd5_2wlyxmbh00000gp/T/xfs-d772c076/user
+# ➤ YN0000: Once you are done run yarn patch-commit -s /private/var/folders/lk/j93xz9pd7nqgd5_2wlyxmbh00000gp/T/xfs-d772c076/user and Yarn will store a patchfile based on your changes.
+# ➤ YN0000: Done in 1s 455ms
+```
+
+Once you have completed your changes, run the command that was output to commit the patch:
+
+```sh
+yarn patch-commit -s /private/var/folders/lk/j93xz9pd7nqgd5_2wlyxmbh00000gp/T/xfs-d772c076/user
+```
+
+This will automatically update the patch file with your changes. Do not forget to run `yarn install` for the patch to be installed and applied to the lockfile.
+
+### Removing a patch
+
+Locate any `patch:` protocol entries in any workspace `package.json` and remove them. The entry will look somewhat similar to: `patch:@changesets/assemble-release-plan@npm%3A5.2.4#~/.yarn/patches/@changesets-assemble-release-plan-npm-5.2.4-2920e4dc4c.patch`, to remove the patch simply extract out the package (everything after the `patch:` and before `#`) and url-decode it and extract the version from it, so for the example it would be `npm:5.2.4`. You should **not** completely remove the line from the `package.json`.
```

Comment:
Do we need the `npm:`? Isn't it the case that we'd only need to specify it if we wanted the package to come from a different registry than the one we use by default, which is `npm`?

I know `yarn` is adding it automatically for patches but when we want to revert to a 'normal' version or resolution, we don't need to specify `npm` and it'll look a bit weird to have some deps specifying it and some not (when not managed by `yarn`).

```suggestion
Locate any `patch:` protocol entries in any workspace `package.json` and remove them. The entry will look somewhat similar to: `patch:@changesets/assemble-release-plan@npm%3A5.2.4#~/.yarn/patches/@changesets-assemble-release-plan-npm-5.2.4-2920e4dc4c.patch`, to remove the patch simply extract out the package (everything after the `patch:` and before `#`) and url-decode it and extract the version from it, so for the example it would be `5.2.4`. You should **not** completely remove the line from the `package.json`.
```

Reply:
not strictly needed no, might be best to remove it, you're right